### PR TITLE
Allow whitespace as trigger characters

### PR DESCRIPTION
**Pull request by NoahTheDuke**
[coc-clojure](https://github.com/noahtheduke/coc-clojure) uses spaces.

### CODE REVIEW

File: src/provider/signatureManager.ts [hunk 1]
```diff
@@ -7,7 +7,12 @@ import { v4 as uuid } from 'uuid'
 export default class SignatureManager extends Manager<SignatureHelpProvider> implements Disposable {
 
   public register(selector: DocumentSelector, provider: SignatureHelpProvider, triggerCharacters?: string[]): Disposable {
-    let characters = triggerCharacters.reduce((p, c) => p.concat(c.split(/\s*/g)), [] as string[])
```
Comment (chemzqm) [inline src/provider/signatureManager.ts:10]: Change this line to
```
let characters = triggerCharacters.reduce((p, c) => p.concat(c.startsWith(' ') ? [' ']: c.split(/\s*/g)), [] as string[])
```

**Review by chemzqm — CHANGES_REQUESTED** [review]

File: src/provider/signatureManager.ts (hunk 1 above)
Comment (NoahTheDuke) [reply src/provider/signatureManager.ts:10]: The goal was to allow for characters other than <space>, such as <newline>. The spec isn't clear either way. What do you think?

Comment (chemzqm) [reply src/provider/signatureManager.ts:10]: Use

``` ts
let characters = triggerCharacters.reduce((p, c) => p.concat(c.length == 1 ? c : c.split(/\s*/g)), [] as string[])
```

Comment (NoahTheDuke) [reply src/provider/signatureManager.ts:10]: Cool, thank you.

File: src/provider/signatureManager.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { v4 as uuid } from 'uuid'
...
+    let characters = triggerCharacters.reduce((p, c) => p.concat(c.length == 1 ? c : c.split(/\s*/g)), [] as string[])
```
Comment (chemzqm) [inline src/provider/signatureManager.ts:10]: Should be `p.concat(c.length == 1 ? [c]`

**Review by chemzqm — CHANGES_REQUESTED** [review]

File: src/provider/signatureManager.ts (hunk 2 above)
Comment (NoahTheDuke) [reply src/provider/signatureManager.ts:10]:
```suggestion
    let characters = triggerCharacters.reduce((p, c) => p.concat(c.length == 1 ? [c] : c.split(/\s*/g)), [] as string[])
```

Comment (NoahTheDuke) [reply src/provider/signatureManager.ts]: Oops, missed your edit. Thanks so much.

**Comment by oblitum** [comment]
@NoahTheDuke Hi. I'm just trying out Clojure for the first time (but I'm a long time coc.nvim user). Does your PR still work in latest coc.nvim? I'm using your coc-clojure extension and this code won't trigger signature help (I was unable to get signatureHelp to be triggered, not sure whether I need to write some special Clojure code declaration that would cause signatures to show up):

```clojure
(defn greet  [name]  (str "Hello, " name) )

(greet <space here triggers nothing>
```

**Comment by oblitum** [comment]
@NoahTheDuke ping?

**Comment by oblitum** [comment]
I'm able to get signatureHelp from clojure-lsp only through manual `CocActionAsync('showSignatureHelp')` from insert mode:

gif-2021-08-28-112332

Automatically after whitespace nothing comes up:
gif-2021-08-28-112540

**Comment by oblitum** [comment]
So, checking this again, clojure-lsp actually doesn't use whitespace as trigger characters for signature help.

**Pull request merged**